test/mocks/buffer/mocks.h

```diff
@@ -8,6 +8,42 @@
 #include "gmock/gmock.h"
 
 namespace Envoy {
+
+class MockBuffer : public Buffer::OwnedImpl {
```

Comment:
same comments about method name casing in this PR (also same comment about picking one of the PRs to get in first and then we can do the other one).